rtt/scripting/StateMachine.cpp

```diff
@@ -47,7 +47,7 @@
 #include <boost/tuple/tuple.hpp>
 #include "internal/mystd.hpp"
 
-#define TRACE_INIT() Logger::In in( _name )
+#define TRACE_INIT() if (mtrace) Logger::In in( _name )
```

Comment:
I agree that some `Logger::In` might be unnecessary and cause a small overhead, but this one is actually wrong: `Logger::In in` must remain valid for the remainder of the function calling `TRACE_INIT()`. With your proposed patch it would be a no-op because `in` immediately goes out of scope again after that line.

I think you would need a patch for the `RTT::Logger::In` class to allow for passing a condition to the constructor. Should be possible in an ABI- and API-compatible way.